### ISSUE TITLE
feature: add nested-struct label (i.e. json label) support in fortio flags to make it extensible

### ISSUE THREAD

**Issue by carolynhu** [fortio/fortio#388]
# Backgroud

Recently, we are adding a bunch of new tests in [istio/tools performance benchmark](https://github.com/istio/tools/tree/master/perf/benchmark), which we need to add a group of new test labels related to Istio Security Policies (like: `numPolicies:10,numSourceIP:10,numPaths:2`). However, we do not want to append those security policy related test labels to the existing test labels (e.g.`"Labels":"613829bb_qps_1000_c_64_1024_none_mtls_both"`) for two reasons: 
(1) The existing test labels are already long enough and our security policy test has several parameters as well, which will make it even longer than the limits what fortio/fortio has (currently is 96 bytes)
(2) The current `Labels` flag support in fortio only supports [string concatenation](https://github.com/fortio/fortio/blob/master/fortio_main.go#L111) which is hard to extend.

# Proposal
Add another test label field, probably being called `-json-label`, in which, user can pass (exactly is to `constuct`) any of those test labels they want, into that new field, and parse those field themselves.

Say, for our use case, our string Label flag would be filled as: "613829bb_abcd", (where `abcd` is our paramID) and we could pass all of our test parameters to `JsonLabel` as:

```
"JsonLabel": {
    "TestID": [  # fill in the actual testID we have in Labels field, here is 613829bb
      {
        "ParamID": abcd,
        "Qps": 1000,
        "Conn": 64,
        "PayloadSize": 1024,
        "SidecarMode": none,
        "TelemetryMode": none_mtls_both,
        "SecurityPolicy": {
            "NumPolicies": 10,
            "NumSourceIP": 10,
            "NumPaths": 2
         }
      }
    ]
}
```

The benefits of the introduced `JsonLabel` (or other name after the discussion) is that:
(1) It simplifies the existing `Labels` field from aggregating all test labels to one place, user only need to save the referenceID in it, and parse those test parameter details in `JsonLabel` by finding the matched referenceID.
(2) It makes the test labels extensible for later use cases.

Also, fortio UI should be updated accordingly.

cc @ldemailly @mandarjog @MichaelEizaguirre

**Comment by ldemailly** [comment]
I'm not opposed to provide a way to add more data to the json, maybe make it extensible but for this particular use case:

Labels are only truncated in the filename, inside the Json they are as long as you wish them to be

could you do something like      `testId,moredetails...` as the labels so even if truncated you have your unique id in front?
(and all the details inside the json)

**Comment by ldemailly** [comment]
also side question: are you using the binary/fortio command line, or the API?
(if using the API you could add whatever you wish to the Json / serialize extended results + context any way you wish)
and if you are using the binary/command line tool; you could also process the json to add more context after said json is produced?

**Comment by carolynhu** [comment]
> I'm not opposed to provide a way to add more data to the json, maybe make it extensible but for this particular use case:
> 
> Labels are only truncated in the filename, inside the Json they are as long as you wish them to be
> 
> could you do something like `testId,moredetails...` as the labels so even if truncated you have your unique id in front?
> (and all the details inside the json)

yes, in my example, the `Labels` is composed of `testID_paramID`, which is the unique id.  By moredetails you mean? you want us to add more details in `Labels`?

**Comment by carolynhu** [comment]
> also side question: are you using the binary/fortio command line, or the API?
> (if using the API you could add whatever you wish to the Json / serialize extended results + context any way you wish)
> and if you are using the binary/command line tool; you could also process the json to add more context after said json is produced?

we are using the binary. Yes, after we got the fortio generated json, we can do the processing we want. But before that, we need to add this kind of support (i mean if we are not interacting with the API).

**Comment by ldemailly** [comment]
apologies this fell through, curious if you still have a need?
